waiter/src/waiter/auth/saml.clj

```diff
@@ -0,0 +1,379 @@
+;;
+;; Copyright (c) Two Sigma Open Source, LLC
+;;
+;; Licensed under the Apache License, Version 2.0 (the "License");
+;; you may not use this file except in compliance with the License.
+;; You may obtain a copy of the License at
+;;
+;;  http://www.apache.org/licenses/LICENSE-2.0
+;;
+;; Unless required by applicable law or agreed to in writing, software
+;; distributed under the License is distributed on an "AS IS" BASIS,
+;; WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+;; See the License for the specific language governing permissions and
+;; limitations under the License.
+;;
+(ns waiter.auth.saml
+  (:require [clj-time.coerce :as tc]
+            [clj-time.core :as t]
+            [clj-time.format :as f]
+            [clojure.data.codec.base64 :as b64]
+            [clojure.java.io :as io]
+            [clojure.tools.logging :as log]
+            [comb.template :as template]
+            [hiccup.core :as hiccup]
+            [hiccup.page]
+            [ring.middleware.params :as ring-params]
+            [ring.util.codec :refer [form-encode url-encode base64-encode]]
+            [ring.util.response :refer [redirect]]
+            [waiter.auth.authentication :as auth]
+            [waiter.middleware :as middleware]
+            [waiter.util.utils :as utils])
+  (:import [javax.xml.parsers DocumentBuilderFactory]
+           [org.opensaml.xml.validation ValidationException]))
```

Comment:
we use lists for class imports.
https://stuartsierra.com/2016/clojure-how-to-ns.html#vectors-or-lists-1

Reply:
done